### ISSUE TITLE
emacsen-common fails on perl5.30 (and warns on 5.18.4+)

### ISSUE THREAD

**Issue by nieder** [fink/fink-distributions#855]
On 11.3 with perl5.30, emacsen-common fails with this error during CompileScript:
```
perl -wc generate-install-list
Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at generate-install-list line 24.
### execution of perl failed, exit code 255
```
On 10.14.5, the above is a warning, and it also gives this warning during install:
```
Setting up emacsen-common (1.4.15-5) ...
defined(@array) is deprecated at /sw/lib/emacsen-common/generate-install-list line 24.
	(Maybe you should just omit the defined()?)
emacsen-common: Handling install of emacsen flavor emacs
...
(finishes installing cleanly)
```
The code looks like this:
```perl
sub installed_add_on_packages_list {
  # Caches value for future use...

  if(! defined(@$installed_add_on_pkgs)) {
    my $oldir = cwd();    
    chdir($lib_dir . "/packages/install/") or die "couldn't chdir";
    @$installed_add_on_pkgs = glob("*");
    chdir($oldir);
  }
  return $installed_add_on_pkgs;
}
```
Following the hint to remove defined(), on 10.14.6, the warning during the build is gone, but the install fails:
```
Setting up emacsen-common (1.4.15-5) ...
Can't use an undefined value as an ARRAY reference at /sw/lib/emacsen-common/generate-install-list line 24.
/sw/bin/dpkg: error processing emacsen-common (--install):
```

tl;dr:
unpatched -> build failure
patched -> install failure

Our emacsen-common is hella old (Debian's current oldest available is 2.0.8 from 2014). Can the perl people take a look to see what's needed to properly work around the current code?

@TheSin- @dmacks

**Comment by nieder** [comment]
@sth0 is suggesting the following patch:
```diff
-  if(! defined(@$installed_add_on_pkgs)) {
+  if( (! defined($installed_add_on_pkgs) ) || ( @$installed_add_on_pkgs == 0 ) ) {
```
This lets it emacsen-common build & install on 12.1

**Comment by sth0** [comment]
I did suggest that a long time ago, and I keep meaning to put it as a PR.  On my todo list...
There was one other weird patch to xinitrc that I needed to get emacs to build on macOS 11.X.

**Comment by nieder** [comment]
For xinitrc, see https://github.com/fink/xinitrc/tree/x11_path
That makes xinitrc work with Xquartz 2.8 (directory change). I'm going to merge that and release when I get back in the New Year.
I'm unaware of any problems with macOS11 and xinitrc.

**Comment by sth0** [comment]
The problem was with a *clean* install of Xquartz 2.8.1 did not create the directory for xinitrc and thus failed in installing.  I can't find it now but all my catalina and greater were failing to build emacs. (Including emacs-nox for some reason).  Having a test and create the directory fo xinitrc with "mkdir -p" fixed it.

An upgrade from Xquartz from version 2.8.0 did not have this problem since the directory was already there.

(all my work systems are down since they have rewired the network and not set it up correctly over the break.)
